lib/matchers/matchers.rb

```diff
@@ -232,7 +232,7 @@
 RSpec::Matchers.define :cmp do |first_expected|
 
   def integer?(value)
-    !(value =~ /\A[1-9]\d*\Z/).nil?
+    !(value =~ /\A[0-9]\d*\Z/).nil?
```

Comment:
Great catch and really good test-case. We left 0 out, to cover the detection for octal numbers. Therefore 0 is only valid if the length of the string is 1